api/v1alpha3/webhooks.go

```diff
@@ -38,10 +38,7 @@ func aggregateObjErrors(gk schema.GroupKind, name string, allErrs field.ErrorLis
 func isValidSSHKey(sshKey *string) field.ErrorList {
 	var allErrs field.ErrorList
 	if sshKey != nil {
-		reg, err := regexp.Compile("[^-A-Za-z0-9-]+")
-		if err != nil {
-			return append(allErrs, field.Invalid(field.NewPath("sshKey"), sshKey, "SSHKey contains invalid character"))
-		}
+		reg := regexp.MustCompile(`^\x00-\x7F]+`)
```

Comment:
Let's make sure we get this right:

Valid SSH key names are the intersection of ISO-8859, ASCII and valid XML entity names.

That gives us a regex of 

```go
var (
  sshKeyValidNameRegex regexp.MustCompile(`^[[:graph:]]+([[:print:]]*[[:graph:]]+)*$`)
)`
```

This should go in a package variable, such that the controller panics as close as possible to startup if the regex is invalid, and that we minimise memory reallocations.

then we can do something like the following (assuming we can change isValidSSHKey's function return  signature from `field.ErrorList` to `error`.

```suggestion
		if sshKey != nil && !sshKeyValidNameRegex.Match([]byte(*sshKey))
		{
		  return field.Invalid(field.NewPath("sshKey"), sshKey, "Name is invalid. Must be specified in ASCII and must not start or end in whitespace")
		}
		return nil
```

Reply:
Make sense..I think the change for error handling needs to be done in a separate PR

Reply:
Yup, that's fine.